src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2Extensions.cs

```diff
@@ -203,5 +243,92 @@ public static ArraySegment<byte> GetArraySegment(this MemoryPoolIterator2 start,
             start.CopyTo(array, 0, length, out length);
             return new ArraySegment<byte>(array, 0, length);
         }
+
+        public static bool GetHttpMethodString(this MemoryPoolIterator2 scan, out string httpMethod)
+        {
+            httpMethod = null;
+            var scanLong = scan.PeekLong();
+
+            if (scanLong == -1)
+            {
+                return false;
+            }
+
+            if (scanLong == _httpConnectMethodLong)
+            {
+                httpMethod = HttpConnectMethod;
+            }
+            else if (((scanLong ^ _httpDeleteMethodLong) << 8) == 0)
+            {
+                httpMethod = HttpDeleteMethod;
+            }
+            else if (((scanLong ^ _httpGetMethodLong) << 32) == 0)
```

Comment:
Super nitpicky, but we might as well check for GETs and POSTs first since I assume they are the most common methods.